### ISSUE TITLE
Problem parsing url with "//" in the path

### ISSUE THREAD

**Issue by daniel-s** [cosimo/perl6-lwp-simple#4]
When parsing a url ( method parse_url() ) with two slashes ( ie. "//" ) parsing the url produces unexpected results:

eg. To create a tiny url that shortens the url http://digg.com
http://tinyurl.com/api-create.php?url=http://digg.com

@path = $url.split(/\/+/);

This will seperate the url into ("http:", "tinyurl.com/api-create.php?url=http:", digg.com)

$path = '/' ~ @path.join('/');

This will join the path back as:
tinyurl.com/api-create.php?url=http:/digg.com

basically, the url parsing doesn't work where there are 2 slashes in the url anywhere after the scheme
